File: Data Structures & Algorithms/invert-a-binary-tree/submission-2.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public TreeNode invertTree(TreeNode root) {
        Queue<TreeNode> queue = new LinkedList<>();
        if(root == null || (root.left == null&&root.right == null)){
            return root;
        }
        TreeNode answer = root;
        queue.add(root);
        while(!queue.isEmpty()){
            TreeNode current = queue.poll();
            TreeNode temp = current.left;
            current.left = current.right;
            current.right = temp;
            if(current.left!=null){
                queue.add(current.left);
            }
            if(current.right!=null){
                queue.add(current.right);
            }            
        }
        return answer;
    }
}

File: Data Structures & Algorithms/three-integer-sum/submission-2.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        std::vector<vector<int>> answer;
        std::sort(nums.begin(), nums.end());
        int left = 0;
        int right = nums.size()-1;
        for(int i = 0; i < nums.size();++i){
            left = i+1;
            right = nums.size()-1;
            if(i != 0){
                if(nums[i] == nums[i-1]){
                    continue;
                }
            }
            while(left<right){
                if(left == i){
                    left++;
                }
                if(right == i){
                    right--;
                }
                if(nums[i]+nums[left]+nums[right]==0){
                    answer.push_back({nums[left], nums[i], nums[right]});
                    left++;
                    right--;
                    while(left<right && nums[left] == nums[left-1]){
                        left++;
                    }
                    while(left<right && nums[right] == nums[right+1]){
                        right--;
                    }
                    
                }
                else if(nums[i]+nums[left]+nums[right]<0){
                    left++;
                }
                else{
                    right--;
                }
            }
        }
        return answer;
    }
};
